impl/maven-impl/src/main/java/org/apache/maven/impl/ImplUtils.java

```diff
@@ -25,18 +25,21 @@
 import java.util.stream.Collectors;
 
 class ImplUtils {
+
+    /**
+     * @deprecated Use {@link Objects#requireNonNull(Object)} instead.
+     */
+    @Deprecated
```

Comment:
We are still in RC. Maybe we can remove it? (or mark it for removal in 4.1?)

Reply:
Something this sure.
But as we are still RC and those methods/classes have been introduced with 4. x, I would even prefer we remove them.

Reply:
agree on less is more, so its gone.